### ISSUE TITLE
story(dialog): dialog element accessibility analysis

### ISSUE THREAD

**Issue by dauriamarco** [lyne-design-system/lyne-components#1369]
### Feature Description

We want to report some considerations regarding the accessibility aspect of the dialog element. Specifically regarding its use in Safari on MacOs:
- any element that is slotted is not read by VoiceOver on Safari
- if the component is inside the Storybook iframe, it works correctly
- sometimes Safari randomly changes behavior/fails

More testing and information will follow shortly on mobile and Windows as well.

### Definition of Done

- [x] Findings are documented (either in this issue or as a markdown file in the repo)

**Comment by dauriamarco** [comment]
A couple of points I forgot to mention:
- one suggestion may be to animate the dialog through `keyframes` rather than overriding the modal's default `display: none` so as to avoid possible problems with accessibility;
- implementing the `backdrop` separately, besides being necessary if you use the `show()` method of the dialog, may even be a better approach since Firefox does not properly support animating the native `::backdrop` element and also creates some problems with using design tokens for styling, given that "it does not inherit from any element and is not inherited from" ([read more](https://stackoverflow.com/questions/58818299/css-variables-not-working-in-dialogbackdrop)).
